x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSet.java

```diff
@@ -253,7 +253,10 @@ private Long dateTime(int columnIndex) throws SQLException {
                     return null;
                 }
                 return JdbcDateUtils.asDateTimeField(val, JdbcDateUtils::asMillisSinceEpoch, Function.identity());
-            };
+            }
+            if (EsType.DATE == type) {
+                return JdbcDateUtils.asDate((String) val).getTime();
```

Comment:
Since the `Date` object is not needed, simply open up `utcMillisRemove` method and use that instead to get the underlying long. Putting that in `Date` and then getting it out in the worse case scenario won't have any effect however its internal Calendar optimization might do.